src/java_bytecode/expr2java.cpp

```diff
@@ -235,6 +237,23 @@ std::string expr2javat::convert_constant(
     dest+='L';
     return dest;
   }
+  else if (src.type() == java_float_type()) {
+    ieee_floatt Fpn(to_constant_expr(src));
+    float F = Fpn.to_float();
+    std::stringstream buffer;
+    buffer << std::hexfloat;
+    buffer << F << 'f';
+    return(buffer.str());
+  }
+   else if (src.type() == java_double_type()) {
+    ieee_floatt Fpn(to_constant_expr(src));
+    double D = Fpn.to_double();
+    std::stringstream buffer;
+    buffer << std::hexfloat;
+    buffer << D;
+    return(buffer.str());
+  }
+  
```

Comment:
Seems a bit clunky but I think that's just what C++ forces you to do.  Other than that; looks good.